WordPress/src/main/java/org/wordpress/android/ui/sitecreation/NewSiteCreationActivity.kt

```diff
@@ -0,0 +1,90 @@
+package org.wordpress.android.ui.sitecreation
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProvider
+import android.arch.lifecycle.ViewModelProviders
+import android.os.Bundle
+import android.support.v4.app.Fragment
+import android.support.v7.app.AppCompatActivity
+import android.view.MenuItem
+import org.wordpress.android.R
+import org.wordpress.android.WordPress
+import org.wordpress.android.analytics.AnalyticsTracker
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.DOMAINS
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.SEGMENTS
+import org.wordpress.android.ui.sitecreation.SiteCreationStep.VERTICALS
+import org.wordpress.android.ui.sitecreation.segments.NewSiteCreationSegmentsFragment
+import org.wordpress.android.ui.sitecreation.segments.SegmentsScreenListener
+import org.wordpress.android.ui.sitecreation.verticals.NewSiteCreationVerticalsFragment
+import org.wordpress.android.ui.sitecreation.verticals.VerticalsScreenListener
+import org.wordpress.android.util.wizard.WizardNavigationTarget
+import javax.inject.Inject
+
+class NewSiteCreationActivity : AppCompatActivity(), SegmentsScreenListener, VerticalsScreenListener {
+    @Inject protected lateinit var viewModelFactory: ViewModelProvider.Factory
+    private lateinit var mainViewModel: NewSiteCreationMainVM
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        (application as WordPress).component().inject(this)
+
+        setContentView(R.layout.site_creation_activity)
+        mainViewModel = ViewModelProviders.of(this, viewModelFactory).get(NewSiteCreationMainVM::class.java)
+        mainViewModel.start()
+
+        if (savedInstanceState == null) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.SITE_CREATION_ACCESSED)
+        }
+        observeVMState()
+    }
+
+    private fun observeVMState() {
+        mainViewModel.navigationTargetObservable
+                .observe(this, Observer { target -> target?.let { showStep(target) } })
```

Comment:
Just sharing my preference: I like to start with `it` in the observer and only assign the name after the null check. It avoids having multiple `target`s or using the `it` at the final stage where it's less meaningful. No change necessary, just sharing :)

```suggestion
                .observe(this, Observer { it?.let { target -> showStep(target) } })
```

Reply:
>It avoids having multiple targets or using the it at the final stage where it's less meaningful.

Hmm, but I have just a single target and I don't use `it` at all, right? Smartcast takes care of casting T? -> T. I like this approach a bit more... but I don't feel strong about it.